include/mxnet/c_api.h

```diff
@@ -405,6 +405,12 @@ MXNET_DLL int MXNDArraySyncCopyFromNDArray(NDArrayHandle handle_dst,
                                            const int i);
 
 /*!
+ * \brief check whether the matrix format is valid
+ * \param full_check if `True`, rigorous check, O(N) operations
+ *    Otherwise basic check, O(1) operations
+ */
+MXNET_DLL int MXNDArrayCheckFormat(NDArrayHandle handle, const bool full_check);
```

Comment:
Do we really need to expose this as a c_api to frontend? I think this should be done silently by the backend

Reply:
@eric-haibin-lin Is it possible to create invalid ndarray?

Reply:
Yes. For example,
```
import mxnet as mx
# Create a CSRNDArray with python lists
shape = (3, 4)
data_list = [7, 8, 9]
indices_list = [0, 2, 1]
indptr_list = [0, 2, 5, 3]
a = mx.nd.sparse.csr_matrix(data_list, indptr_list, indices_list, shape)
# Inspect the matrix
print a.asnumpy()
```
It is an invalid csr ndarray and will cause segfault in my machine (pip install mxnet-cu80 --pre).

Reply:
I agree with you that it's better to check in backend silently. But as `sparse. csr_matrix` is in python, I didn't figure out a good way to add check_format into backend. Maybe we could put it into `sparse. csr_matrix`, not directly exposed to users?

Reply:
@eric-haibin-lin Why would this fail?

Reply:
Yes it's possible to create an invalid csr matrix. e.g. in scipy
```
>>> a = spsp.csr_matrix(([0],[10],[0,1]), shape=(1,2))
>>> a
<1x2 sparse matrix of type '<type 'numpy.int64'>'
	with 1 stored elements in Compressed Sparse Row format>
>>> a.check_format()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/scipy/sparse/compressed.py", line 179, in check_format
    (minor_name,minor_dim))
ValueError: column index values must be < 2
```

Can we implemented it in backend as a util function, and wrap it as an operator to expose it to frontend? 

@piiswrong Do you suggest we inspect the data every time a CSRNDArray is constructed?? I think that's very expensive.